### PR TITLE
Adds catalog item example to externalized-configuration.md

### PR DESCRIPTION
**Pull request by aledsage**

### CODE REVIEW

File: brooklyn-docs/guide/ops/externalized-configuration.md [hunk 1]
```diff
@@ -122,6 +122,19 @@ brooklyn.location.jclouds.aws-ec2.credential=$brooklyn:external("mysupplier", "a
...
+
+    brooklyn.catalog:
+      id: com.example.myblueprint
+      version: 1.2.3
+      brooklyn.libraries:
+      - $brooklyn:external("mysupplier", "my-blueprint-lib")
+      item:
+        type: com.example.MyBlueprint
+
+
```
Comment (alasdairhodge) [inline brooklyn-docs/guide/ops/externalized-configuration.md]: Good thing to demonstrate,but example feels a bit contrived. Perhaps show a *location* catalog item, with user ID and credential externalised?

File: brooklyn-docs/guide/ops/externalized-configuration.md [hunk 2]
```diff
@@ -122,6 +122,23 @@ brooklyn.location.jclouds.aws-ec2.credential=$brooklyn:external("mysupplier", "a
...
+      brooklyn.libraries:
+      - >
+        $brooklyn:formatString("https://%s:%s@repo.example.com/libs/myblueprint-1.2.3.jar", 
+        external("mysuppier", "username"), external("mysupplier", "password"))
+      item:
+        type: com.example.MyBlueprint
+
+Note the `>` in the example above is used to split across multiple lines.
+
+
```
Comment (alasdairhodge) [inline brooklyn-docs/guide/ops/externalized-configuration.md]: Much better example. I didn't realise that the `$brooklyn` prefix can be dropped inside the scope of the function arguments; does this definitely parse? ;o)

File: brooklyn-docs/guide/ops/externalized-configuration.md [hunk 3]
```diff
@@ -122,6 +122,23 @@ brooklyn.location.jclouds.aws-ec2.credential=$brooklyn:external("mysupplier", "a
 {% endhighlight %}
 
 
+## Referring to External Configuration in catalog items
```
Comment (alasdairhodge) [inline brooklyn-docs/guide/ops/externalized-configuration.md]: Capitalisation of "catalog items" is inconsistent with other section headings. /nitpick

**Comment by aledsage** [comment]
Thanks @alasdairhodge - have changed heading capitalisation. And yes, the example parses :-)
Merging now.

**Pull request merged**